ocw/lib/EC2.py

```diff
@@ -334,3 +355,42 @@ def cleanup_images(self, valid_period_days):
                     else:
                         self.log_dbg("Delete image '{}' (ami:{})".format(img['Name'], img['ImageId']))
                         self.ec2_client(region).deregister_image(ImageId=img['ImageId'], DryRun=False)
+
+    def cleanup_k8s_jobs(self):
+        self.create_credentials_file()
+        self.cmd_exec("aws sts get-caller-identity")
+
+        clusters = {}
+        for region in self.cluster_regions:
+            response = self.eks_client(region).list_clusters()
+            if 'clusters' in response and len(response['clusters']) > 0:
+                clusters[region] = []
+                for cluster_name in response['clusters']:
+                    client = self.kubectl_client(region, cluster_name)
+                    if client is not None:
+                        now = datetime.now(timezone.utc)
+
+                        ret = client.list_job_for_all_namespaces(watch=False)
+                        for job in ret.items:
+                            age = (now - job.status.start_time).days
+                            if age > 1:
+                                if not self.dry_run:
+                                    self.log_info(f"Deleting from {cluster_name} the job {job.metadata.name} with age {age}")
+                                    client.delete_namespaced_job(job.metadata.name, job.metadata.namespace)
+                                else:
+                                    self.log_info(f"Skip deleting from {cluster_name} the job {job.metadata.name} with age {age}")
+
+    def create_credentials_file(self):
```

Comment:
suggestion: 

```
def create_credentials_file(self):
        directory = "/root/.aws"
        file = f"{directory}/credentials"      
        if not os.path.exists(directory):
            os.mkdir(directory)
            with open("/root/.aws/credentials", "w", encoding="utf8") as file:
                 file.write("[default]\n")
                 file.write(f"aws_access_key_id={self.__key}\n")
                 file.write(f"aws_secret_access_key={self.__secret}\n")
        if self.cmd_exec("aws sts get-caller-identity") != 0 :
            raise botocore.exceptions.ClientError   # this part is just a stub . you need to find how to properly throw ClientError 
```

Reply:
I did this in a different way to be sure that "aws sts get-caller-identity" because the credentials already exists but are not valid any longer.

Reply:
ok your version looks good